File: kotlin/programming-kotlin_code/spring/java/todo/src/main/java/com/agiledeveloper/todo/TaskService.java
/***
 * Excerpted from "Programming Kotlin",
 * published by The Pragmatic Bookshelf.
 * Copyrights apply to this code. It may not be used to create training material,
 * courses, books, articles, and the like. Contact us if you are in doubt.
 * We make no guarantees that this code is fit for any purpose.
 * Visit http://www.pragmaticprogrammer.com/titles/vskotlin for more book information.
***/
//Java code only for comparison purpose
package com.agiledeveloper.todo;

import java.util.Optional;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;
            
@Service
@Transactional
class TaskService {
  private final TaskRepository repository;
  
  public TaskService(TaskRepository repository) {
    this.repository = repository;
  }
  
  Iterable<Task> getAll() {
    return repository.findAll();
  }

  Task save(Task task) {
    return repository.save(task);
  }
  
  boolean delete(Long id) {
    boolean found = repository.existsById(id);
    
    if (found) {
      repository.deleteById(id);
    }

    return found;
  }
}

File: kotlin/programming-kotlin_code/spring/kotlin/todo/src/main/kotlin/com/agiledeveloper/todo/TaskService.kt
package com.agiledeveloper.todo

import org.springframework.stereotype.Service
import org.springframework.transaction.annotation.Transactional
            
@Service
@Transactional
class TaskService(val repository: TaskRepository) {
  fun getAll() = repository.findAll()
  
  fun save(task: Task) = repository.save(task)
  
  fun delete(id: Long):  Boolean {
    val found = repository.existsById(id)
    
    if (found) {
      repository.deleteById(id)
    }
    
    return found
  }
}
